assets/js/monitor.js

```diff
@@ -57,40 +57,57 @@ jQuery(async function () {
 
 function updateEnvironmentList() {
   const validEnvironments = ['prod', 'preprod', 'test', 'stage', 'dev']
+  const selectedEnvironments = []
+  let showProbation = false
+  let showPrisons = false
+  let showStatusUp = false
+  let showStatusDown = false
+  let showStatusMissing = false
+
+  $('.environments .govuk-checkboxes__input:checked').each((index, e) => {
+    const environment = $(e).val()
 
-  $('#statusRows tr').show()
+    if (validEnvironments.includes(environment)) {
+      selectedEnvironments.push(environment)
+    }
+  })
 
-  if (!$('#status-up').is(':checked')) {
-    $('.statusTileUp').hide()
+  if ($('#status-up').is(':checked')) {
+    showStatusUp = true
   }
-  if (!$('#status-down').is(':checked')) {
-    $('.statusTileDown').hide()
+
+  if ($('#status-down').is(':checked')) {
+    showStatusDown = true
   }
-  if (!$('#status-missing').is(':checked')) {
-    $('#statusRows tr').each(function () {
-      if (!$(this).hasClass('statusTileDown') && !$(this).hasClass('statusTileUp')) {
-        $(this).hide()
-      }
-    })
+
+  if ($('#status-missing').is(':checked')) {
+    showStatusMissing = true
   }
+
   if ($('#hmpps-area-probation').is(':checked')) {
-    $('#statusRows tr').each(function () {
-      $(this).data('probation') ? $(this).show() : $(this).hide()
-    })
+    showProbation = true
   }
+
   if ($('#hmpps-area-prisons').is(':checked')) {
-    $('#statusRows tr').each(function () {
-      $(this).data('prisons') ? $(this).show() : $(this).hide()
-    })
+    showPrisons = true
   }
 
-  $('.environments .govuk-checkboxes__input:not(:checked)').each((index, e) => {
-    const environment = $(e).val()
-
-    if (validEnvironments.includes(environment)) {
-      $(`.${environment}`).hide()
-    }
-  })
+  $('#statusRows tr')
+    .hide()
```

Comment:
Hide everything, then loop through each row showing it if it matches the filters